### PR TITLE
XW-4310 | eslint ember plugin

### PR DESCRIPTION
**Pull request by hakubo**
* init arrays and objects for objects inside init hook
* don't use sendAction
* passing actions refactor

### CODE REVIEW

**Review by bkoval — APPROVED** [review]

File: app/components/application-wrapper.js [hunk 1]
```diff
@@ -188,8 +188,8 @@ export default Component.extend({
 			 * pass it up to handleLink
 			 */
 			if (!target.href.match(`^${window.location.origin}/a/.*/comments$`)) {
-				this.sendAction('closeLightbox');
-				this.sendAction('handleLink', target);
+				this.get('closeLightbox')();
```
Comment (latata) [inline app/components/application-wrapper.js:191]: Don't we loose `this` context is such execution? Or ember binds it correctly?

File: app/components/article-featured-video.js [hunk 2]
```diff
@@ -28,20 +28,15 @@ export default Component.extend(RenderComponentMixin, {
...
+	placeholderImage: readOnly('model.embed.jsParams.playlist.0.image'),
 	placeholderStyle: computed('placeholderImage', function () {
 		return htmlSafe(`background-image: url(${this.get('placeholderImage')})`);
 	}),
 
-	// when navigating from one article to another with video, we need to destroy player and
-	// reinitialize it as component itself is not destroyed. Could be done with didUpdateAttrs
-	// hook, however it is fired twice with new attributes.
-	videoIdObserver: on('didInsertElement', observer('model.embed.jsParams.videoId', function () {
+	didUpdateAttrs() {
```
Comment (latata) [inline app/components/article-featured-video.js:38]: Isn't it fired twice?

File: app/templates/file.hbs [hunk 3]
```diff
@@ -2,14 +2,13 @@
...
 	{{#article-wrapper
-		articleRendered='articleRendered'
 		displayEmptyArticleInfo=false
 		model=model
-		openLightbox='openLightbox'
 		showComments=false
 		siteName=siteName
-		toggleDrawer='toggleDrawer'
 		wikiPageHeaderSubtitle=(i18n 'file-page.subtitle')
+		openLightbox=(action 'openLightbox' target=application)
```
Comment (latata) [inline app/templates/file.hbs:10]: do we keep alphabetical order in component properties? Lint it! ;)

**Review by latata — APPROVED** [review]

File: app/components/article-featured-video.js (hunk 2 above)
Comment (hakubo) [reply app/components/article-featured-video.js:38]: nope, I checked it and it fired only once

File: app/templates/file.hbs (hunk 3 above)
Comment (hakubo) [reply app/templates/file.hbs:10]: not yet, but I think we will :)

File: app/components/application-wrapper.js (hunk 1 above)
Comment (hakubo) [reply app/components/application-wrapper.js:191]: nope, Ember handles it like it should
https://ember-twiddle.com/55e68273e4e681d56a131b88f11cca3e?openFiles=controllers.application.js%2Ctemplates.components.my-component.hbs

**Pull request merged**